### PR TITLE
Add CI and fix issues

### PR DESCRIPTION
**Pull request by davfsa**

### CODE REVIEW

File: bannou/database/tags.py [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
 
 class Tag(ormar.Model):
-    class Meta(BaseMeta):
+    class Meta(base.BaseMeta):
         tablename = "tags"
 
     id: int = ormar.Integer(primary_key=True)
     # TODO: created_by column mapping to an individual user via User model
-    created_on: datetime.datetime = ormar.DateTime(nullable=False, server_default=sqlalchemy.func.now())
+    created_on: datetime.datetime = ormar.DateTime(nullable=True, server_default=sqlalchemy.func.now())
```
Comment (davfsa) [inline bannou/database/tags.py:22]:
```suggestion
    created_on: datetime.datetime = ormar.DateTime(nullable=False, server_default=sqlalchemy.func.now())
```

File: alembic.ini [hunk 2]
```diff
@@ -8,7 +8,7 @@ script_location = scripts/migrations
 # Uncomment the line below if you want the files to be prepended with date and time
 # see https://alembic.sqlalchemy.org/en/latest/tutorial.html#editing-the-ini-file
 # for all available tokens
-file_template = %%(year)d_%%(month).2d_%%(day).2d_%%(hour).2d%%(minute).2d_%%(slug)s
+file_template = %%(year)d_%%(month).2d_%%(day).2d_%%(hour).2d_%%(minute).2d_%%(second).2d.%%(rev)s
```
Comment (null-domain) [inline alembic.ini:11]: Any reason we use the revision hash in the filename as opposed to the slug? IMO slug is better to provide an 'at a glance' description of what the revision does without having to open it.

File: bannou/settings.py [hunk 3]
```diff
@@ -98,7 +98,7 @@ def port(self) -> int:
     def path(self) -> str:
         return self._path
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return (
             "DSN("
```
Comment (null-domain) [inline bannou/settings.py:103]: I forgot to change this whenever I made the PR to rename DSN to Service, but the beginning of the repr here should be changed to match.

File: config.example.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      class: logging.StreamHandler
+      stream: ext://sys.stdout
+
+    # rfh:
+    #   level: INFO
+    #   formatter: file
+    #   class: logging.handlers.RotatingFileHandler
+    #   filename: output.log
+    #   maxBytes: 1048576
+    #   backupCount: 5
```
Comment (null-domain) [inline config.example.yaml:38]: Why is it commented out by default?

File: dev-requirements.txt [hunk 5]
```diff
@@ -1,4 +1,2 @@
-black==23.1.0
-isort==5.12.0
-ruff==0.0.256
-slotscheck==0.16.5
\ No newline at end of file
+nox==2022.11.21
+alembic[tz]==1.10.2
```
Comment (null-domain) [inline dev-requirements.txt:2]: Alembic would be needed in a production environment, not a development environment, unless you're making bigbrain moves I can't see yet

File: scripts/migrations/env.py [hunk 6]
```diff
@@ -1,35 +1,39 @@
...
 from logging.config import fileConfig
 
+import sqlalchemy
 from alembic import context as alembic_context
-from sqlalchemy import engine_from_config, pool
-from sqlalchemy.engine import Connection
-from sqlalchemy.ext.asyncio import AsyncEngine
+from sqlalchemy import engine as sqlachademy_engine
+from sqlalchemy import pool as sqlachademy_pool
+from sqlalchemy.ext import asyncio as sqlachademy_asyncio
```
Comment (null-domain) [inline scripts/migrations/env.py:12]: `sqlalchemy` not `sqlachademy`

**Review by null-domain — CHANGES_REQUESTED** [review]
What's the plan for dealing with the failing slotscheck tests?

**Comment by null-domain** [comment]
Noting the issues you're facing with using ormar (and I'm spotting a few minor faults that, while not huge issues, are still just annoying) - would it be worth ditching ormar in its entirety and just [using SQLAlchemy models directly](https://docs.sqlalchemy.org/en/20/orm/quickstart.html#declare-models)?

Downside of doing so is we lose the cleanliness of ormar models, plus the strong-typing and validation of them (since ormar models are based on Pydantic models underneath), but the upside would be that we could have stronger control over how the models work.

**Comment by null-domain** [comment]
Merging this for the time being so progress can continue. We can revisit in the future (particularly to sort out the slotscheck issue) and see what else needs to change.

**Review by null-domain — APPROVED** [review]

**Pull request merged**